src/plugins/data/public/plugin.ts

```diff
@@ -20,7 +20,7 @@
 import { PluginInitializerContext, CoreSetup, CoreStart, Plugin } from '../../../core/public';
 import { ExpressionsService } from './expressions/expressions_service';
 
-export class DataPublicPlugin implements Plugin<{}> {
+export class DataPublicPlugin implements Plugin<ReturnType<DataPublicPlugin['setup']>, void> {
```

Comment:
Just a note that I'm working on some conventions for the plugin files here which would include exporting a dedicated `DataSetup` interface that you could pass to this generic: #39554 

Would be interested in your thoughts on that PR if you have time to review!